Design/spectrum_number_and_workspace_index_abstraction.md

```diff
@@ -0,0 +1,269 @@
+# Indexing Abstraction Layer
```

Comment:
@FedeMPouzols I consider the term "workspace index" confusing (after all it is not an index of a workspace in, say, a list of workspaces), so this was just an attempt to sneak in alternative terminology. But maybe we just have to stick with the old one?

My intention was to use "spectrum index" instead of "workspace index", i.e., it is just a memory offset and has nothing to do with working in NeXus files.